repeater/src/main/scala/com.snowplowanalytics.snowplow.storage.bigquery.repeater/PayloadParser.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2019 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.storage.bigquery.repeater
+
+import cats.syntax.either._
+import cats.data.NonEmptyList
+import io.circe._
+import io.circe.syntax._
+import io.circe.generic.semiauto._
+import com.snowplowanalytics.snowplow.analytics.scalasdk.Event
+import BadRow.{ParsingError, EventParsingError}
+
+/**
+  * Parser which is used for reconverting failedInsert payload into
+  * enriched event. Due to some reasons, payload can not be reconverted
+  * into event fully however it can be converted mostly. In order to
+  * get more information, look at the comments of "ReconstructedEvent"
+  */
+object PayloadParser {
+  def parse(payload: JsonObject): Either[ParsingError, ReconstructedEvent] =
+    payload
+      .add("contexts", Json.obj())
+      .add("derived_contexts", Json.obj())
+      .add("unstruct_event", Json.obj())
+      .asJson.as[Event]
+      .leftMap { e =>
+        ParsingError(
+          payload.asJson.noSpaces,
+          NonEmptyList.of(EventParsingError(e.message, e.history.foldLeft(List[String]())((l, i) => i.toString::l)))
+        )
+      }
+      .map(ReconstructedEvent(_, getSelfDescribingEntities(payload)))
+
+  private def getSelfDescribingEntities(payload: JsonObject): List[SelfDescribingEntity] = {
+    val jsonPrefixes = List("contexts", "unstruct_event")
```

Comment:
Why `derived_contexts` is not among these?

Reply:
As far as my observation, 'derived_contexts' is prefixed with `contexts` in the shredded type. Therefore, there is no need to check whether they are prefixed with `derived_contexts` or not. I think I should mentioned in the comment.